pkg/cable/libreswan/libreswan.go

```diff
@@ -103,10 +124,11 @@ type specification struct {
 	PSKSecret   string
 	LogFile     string
 	NATTPort    string `default:"4500"`
+	AuthMode    string `default:"psk"` // Authentication mode: "psk" or "cert"
 }
 
 // NewLibreswan starts an IKE daemon using Libreswan and configures it to manage Submariner's endpoints.
-func NewLibreswan(localEndpoint *submendpoint.Local, _ *types.SubmarinerCluster) (cable.Driver, error) {
+func NewLibreswan(syncerConfig *broker.SyncerConfig, localEndpoint *submendpoint.Local, _ *types.SubmarinerCluster) (cable.Driver, error) {
```

Comment:
I think we should make this more generic by passing the `SigningRequestor` (instantiated in the `gateway`) to the cable engine/drivers.